### PR TITLE
Custom error pages

### PR DESCRIPTION
**Pull request by defong**
### Context
The need for custom error pages.

### Changes proposed in this pull request
Added redirects to customer pages 

### Guidance to review
https://localhost:44364/error/401
https://localhost:44364/error/403
https://localhost:44364/error/404
https://localhost:44364/error/500

# After

screen shot 2018-07-12 at 15 50 16
screen shot 2018-07-12 at 15 50 10
screen shot 2018-07-12 at 15 50 05
screen shot 2018-07-16 at 10 50 03

### CODE REVIEW

**Comment by fofr** [comment]
Could you include some screenshots please?

**Comment by dankmitchell** [comment]
Please see attached @fofr

**Comment by defong** [comment]
Thanks @dankmitchell for the screenshots

**Comment by dankmitchell** [comment]
Cheers @fofr! Chasing a code review now :)

File: src/ui/Views/Auth/AccessDenied.cshtml [hunk 1]
```diff
@@ -1,20 +0,0 @@
-<div class="grid-row">
-  <div class="column-two-thirds">
-    <h1 class="heading-xlarge">
-      We don’t know which organisation you’re part of
-    </h1>
-
-    <p>You have successfully signed in with your DfE account but we don’t recognise the email address you’ve used. This might have happened if you were forwarded the original invitation email or if you’ve recently updated your email address.</p>
-    <p>We need more information from you before we can give you access to your organisation’s courses.</p>
```
Comment (timabell) [inline src/ui/Views/Auth/AccessDenied.cshtml]: We'll have to figure out another way of displaying this info before merging this I think.  I agree it shouldn't be the blanket 401 text, that was just a temporary hack.

Comment (benilovj) [reply src/ui/Views/Auth/AccessDenied.cshtml]: Agreed we can't lose the functionality of showing this copy when a user is authenticated but unauthorised.

Comment (dankmitchell) [reply src/ui/Views/Auth/AccessDenied.cshtml]: Okay, so we have replaced this specific text with `This page is restricted to certain users only.` but if we want different copy for each use case then we need to make this text dynamic so when we raise this exception we pass in the message 👍

**Comment by timabell** [comment]
```
An unhandled exception occurred while processing the request.
InvalidOperationException: The view 'AccessDenied' was not found. The following locations were searched:
/Views/Auth/AccessDenied.cshtml
/Views/Shared/AccessDenied.cshtml

Microsoft.AspNetCore.Mvc.ViewEngines.ViewEngineResult.EnsureSuccessful(IEnumerable<string> originalLocations)
```

**Comment by defong** [comment]
@timabell fixed broken error

**Comment by timabell** [comment]
* User with no mc_users entry is getting the original message now.
* random url is not showing the 404 page

**Pull request merged**